### ISSUE TITLE
Make `version` on element level optional (already there for local, required for external resources)

### ISSUE THREAD

**Issue by morri-son** [open-component-model/ocm#1173]
**What would you like to be added**:
Although specified as`optional` in the [OCM Specification for the element identity](https://github.com/open-component-model/ocm-spec/blob/main/doc/01-model/03-elements-sub.md#element-identity), it is required to specify a version on  resource level when not describing local resources, but an access to an external resource. For local resources it can be omitted and gets implicitly the version defined on component level.

It should not be required to set a version on element level, regardless if for a local resource of for one described by an external access. 

**Why is this needed**:
It is a bit confusing why the version is required for a subset for resources in the component constructor.

**Comment by jakobmoellerdev** [comment]
how is this a blocker? just specify `v0.0.0` as version no?

**Comment by akafazov** [comment]
Yes, you are right. I was using the same version in the `imageReference` tag as well, but see those can be different.

**Comment by matthiasbruns** [comment]
Sprint hygiene: automatically moved from **Sprint 65 - Pipeline Pandemonium** to **Sprint 66 - Technical Debt Thunderdome** because Needs Refinement items are planned in the next sprint.
